lib/logstash/inputs/rabbitmq.rb

```diff
@@ -93,15 +97,38 @@ def declare_queue
       def consume!
         # we manually build a consumer here to be able to keep a reference to it
         # in an @ivar even though we use a blocking version of HB::Queue#subscribe
-        @consumer = @hare_info.queue.build_consumer(:block => true) do |metadata, data|
-          @codec.decode(data) do |event|
-            decorate(event)
-            @output_queue << event if event
+
+        # The logic here around resubscription might seem strange, but its predicated on the fact
+        # that we rely on MarchHare to do the reconnection for us with auto_reconnect.
+        # Unfortunately, while MarchHare does the reconnection work it won't re-subscribe the consumer
```

Comment:
Blocking consumers are expected to unblock before recovery. Using a blocking consumer makes certain sense in small scripts but I see no justification for doing it in Logstash.